a3p-integration/proposals/a:upgrade-14/testRepairs.ts

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env tsx
+
+import { execFileSync } from 'child_process';
+import { readFile, writeFile } from 'fs/promises';
+
+import { makeAgd } from '@agoric/synthetic-chain/src/lib/agd-lib.js';
+import {
+  getUser,
+  voteLatestProposalAndWait,
+} from '@agoric/synthetic-chain/src/lib/commonUpgradeHelpers.js';
+import assert from 'assert';
+
+const SUBMISSION_DIR = 'invite-submission';
+
+const staticConfig = {
+  deposit: '10000000ubld', // 10 BLD
+  installer: 'gov1', // as in: agd keys show gov1
+  proposer: 'validator',
+  collateralPrice: 6, // conservatively low price. TODO: look up
+  swingstorePath: '~/.agoric/data/agoric/swingstore.sqlite',
+};
+
+/** Provide access to the outside world via context. */
+const makeContext = async () => {
+  const config = {
+    chainId: 'agoriclocal',
+    ...staticConfig,
+  };
+
+  const agd = makeAgd({ execFileSync }).withOpts({
+    keyringBackend: 'test',
+  });
+
+  return { agd, config };
+};
+
+// XXX vestige of Ava
+const step = async (name: string, fn: Function) => {
+  console.log(name);
+  await fn();
+};
+
+const replaceAddressInFile = async (string, fileName, replacement) => {
+  const scriptBuffer = await readFile(`${SUBMISSION_DIR}/${fileName}.tpl`);
```

Comment:
Nit: the order of arguments is surprising. Also why hard code `SUBMISSION_DIR` instead of expecting it in the input `fileName`?

Reply:
good suggestions.